commitizen/hooks.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from commitizen import cmd, out
+from commitizen.exceptions import RunHookError
+
+
+def run(hooks, _env_prefix="CZ_", **env):
+    if isinstance(hooks, str):
+        hooks = [hooks]
+
+    for name, value in env.items():
```

Comment:
The hooks part look good to me. But I'm a bit confused here. Why do we need this functionality?

Reply:
These are environment variables, available during executing of the scripts/commands in the hook. So for instance, the post bump hook passes these kind of environment variables to the script:
```bash
CZ_POST_CURRENT_TAG_VERSION=v3.0.0
CZ_POST_CHANGELOG_FILE_NAME=CHANGELOG.md
CZ_POST_WAS_INITIAL=True
CZ_POST_CURRENT_VERSION=3.0.0
CZ_POST_INCREMENT=MAJOR
CZ_POST_MESSAGE=bump: version 2.38.0 → 3.0.0
CZ_POST_PREVIOUS_VERSION=2.38.0
CZ_POST_PREVIOUS_TAG_VERSION=v2.38.0
```
Depending on what it is you are running during these hooks, these variables can be very useful.

Reply:
I'm not sure I like this idea. (nor did i strongly oppose it) @woile what do you think?

Reply:
Is it the fact that the environment variables are set "globally" instead of as argument to `subprocess.Popen` or are you just not sure you like environment variables here at all? It is a relatively flexible way to pass metadata to the hook scripts and I think it is important to be able to pass that metadata in _some_ way to the hook scripts.

In the case of the `subprocess.Popen` argument, I could refactor it a bit so the env is not set directly to `os.environ` but rather passed to subprocess as argument.

Reply:
I think what I'm not sure about is setting environment variables globally. It could potentially pollute or be polluted by other stuffs. But the `subprocess.Popen` solution sounds good!

Reply:
I've implemented that and added a testcase for the pure function I introduced for it.